pymeasure/instruments/instrument.py

```diff
@@ -81,15 +84,21 @@ def id(self):
             return self.adapter.ask("*IDN?").strip()
         else:
             return "Warning: Property not implemented."
+    
+    @property
+    def id_dict(self):
+        return dict(zip(("vendor", "model", "serial", "firmware"), self.id.split(",")))
```

Comment:
This assumes a specific format from the `id`, which does not generalize for different adapters. Those cases need to be handled in this class, or it should be removed.

Reply:
Isn't the id string standardized?